### ISSUE TITLE
Firefox's own Dark Mode setting not honoured by AMO - blindingly bright background

### ISSUE THREAD

**Issue by Jackalaq** [mozilla/addons#2193]
As description.  Dark Mode is selected in Settings and even with Dark Reader AddOn installed, the page is disturbingly bright.

Firefox for Linux Mint

Version 115

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDFRNT-314)

**Comment by i1li** [comment]
The year is 2024, why does mozilla block the most basic of accessibility options, choosing instead a path of able-bodied supremacism, and white supremacism? Vision sensitivity isn't an obscure issue, does mozilla even have a dev team? Heres an example of how to honor user accessibility preferences with just a few lines of code, happy to assist with implementation: https://github.com/i1li/i1li.github.io/blob/ed2f57e824a8ad15dba59b5c4217dd06f9e1a8b2/js/script.js#L35

**Comment by diox** [comment]
AMO team is small and we have a limited bandwidth, we simply can't do everything. The good news is that this is open-source, so if you want to contribute a patch, feel free to submit a pull request over at https://github.com/mozilla/addons-frontend/pulls.

**Comment by fiendUnpleasant** [comment]
@diox "The good news is that this is open-source," is a cop out. What happens when I submit a patch that is the code as it currently stands with background=black and text=white? Will it be accepted or will you just deny the patch of the disabled person just trying get by day to day

**Comment by diox** [comment]
The previous commenter said he was happy to assist, so my reply was me trying to take them up on their offer.

I would gladly review a patch that implements this. It's not trivial however, there is a lot more than just one color to change. If someone really wants to give it a try rather than posting more sarcastic or aggressive comments, I have a [WIP branch](https://github.com/diox/addons-frontend/tree/dark-mode) (that I worked on during my personal time, btw, because I *do* care about this issue) that could be used as a starting point.

**Comment by fiendUnpleasant** [comment]
@diox Hey sorry for being a dick I was in the middle of a migraine exacerbated by this issue. It's difficult to remain civil when it feels like someone is pushing an ice pick into the front of your skull. I am working with a therapist. I am not trying to excuse anything but I was wanting to have something to say for myself other than "sorry I am a dick"

**Comment by nothingrandom** [comment]
Assign this to me 🙂

**Comment by ioanarusiczki** [comment]
I don't see differences between Chrome - Firefox - Android using Dark (on FF I also tried with Automatic while the Dark Theme is enabled from about:addons or [Dark Reader](https://addons.mozilla.org/en-US/firefox/addon/darkreader/))

What I've noticed today

Collections -> view a collection 

<img width="1777" height="863" alt="Image" src="https://github.com/user-attachments/assets/be39cef3-4ec4-49f6-81fd-96376e8fd66d" />

Some txt black - still has decent contrast on https://addons-dev.allizom.org/en-US/about

<img width="1312" height="305" alt="Image" src="https://github.com/user-attachments/assets/2393a8f2-5f4f-420f-8320-ea928e514a4c" />

This is at the bottom of abuse reports page https://addons-dev.allizom.org/en-US/firefox/feedback/addon/scenario1/

<img width="1711" height="555" alt="Image" src="https://github.com/user-attachments/assets/3b55de80-fe52-47a8-af82-ff34a97db6fc" />

Desktop view https://addons-dev.allizom.org/en-US/android/ blurry urls (imo they're visible enough)

<img width="1697" height="111" alt="Image" src="https://github.com/user-attachments/assets/4731f749-5c75-4532-9679-dde4575897cf" />

Collections page https://addons-dev.allizom.org/en-US/firefox/collections/

<img width="1877" height="863" alt="Image" src="https://github.com/user-attachments/assets/a924807c-6371-4b67-a848-029db639543c" />

Custom url field at create collection 

<img width="1887" height="548" alt="Image" src="https://github.com/user-attachments/assets/547a74d4-52c1-4917-a3d5-7c6e426edc9b" />

Edit profile page Email Address field

<img width="1072" height="275" alt="Image" src="https://github.com/user-attachments/assets/7794e0af-ca56-4fb6-9200-1d7cd0fe79c9" />

Grey txt on edit profile page but imo it's fine,  with Dark Reader enabled this would look a bit brighter

<img width="1098" height="607" alt="Image" src="https://github.com/user-attachments/assets/f3d822db-2c8c-48aa-9c8a-84083dd93191" />

While making a rating - the "Saving star rating" and "Star rating saved" 
<img width="662" height="269" alt="Image" src="https://github.com/user-attachments/assets/4d14e7be-26c3-4232-b116-1d01e0f5a25c" />

Remove button after Install -> but while hovering over there's a clearer contrast
<img width="1687" height="596" alt="Image" src="https://github.com/user-attachments/assets/5256f5c4-5d30-47d5-b96c-c260d0629c11" />

review guidelines url at edit rating which at hover is visible because it's blue

<img width="1071" height="230" alt="Image" src="https://github.com/user-attachments/assets/9da54153-e67e-4a55-b992-8d3d43a8a130" />

the paginator (looks like this on all pages) + themes section on view profile page - with some white stripes

<img width="1130" height="992" alt="Image" src="https://github.com/user-attachments/assets/c9b38812-bde3-4d0f-80dd-cb8a0220ff88" />

With the Automatic option while Dark Reader is enabled instead of the Dark Theme from about:addons I see most of the above looking good. 

One thing would look weird - addon star rating

<img width="1070" height="177" alt="Image" src="https://github.com/user-attachments/assets/d4c5d123-df7d-43ae-84a8-330241a32e21" />

But collection pages look good,  or the pagination is dark

<img width="1653" height="550" alt="Image" src="https://github.com/user-attachments/assets/a670bba2-9fcd-4877-b6ff-1ef8e090da1a" />
<img width="1726" height="790" alt="Image" src="https://github.com/user-attachments/assets/d8dc4d16-72cf-4b8c-814f-0890d3975ec3" />
<img width="1132" height="631" alt="Image" src="https://github.com/user-attachments/assets/d1875904-b4fd-4133-848c-4f04969e9f5c" />
<img width="1103" height="172" alt="Image" src="https://github.com/user-attachments/assets/f91b9546-38ed-4fda-be57-7e75d8f79e9c" />

**Comment by acornestean** [comment]
With the default Dark Theme enabled in about:addons + Automatic option on AMO **OR** just Dark option on AMO.

- Parts of the "My collections" padding does not change color.
- "This add-on is not actively monitored for security by Mozilla. Make sure you trust it before installing." banner does not change color between Light or Dark modes. It stays the same no matter what.

<img width="1911" height="989" alt="Image" src="https://github.com/user-attachments/assets/d2831af5-ddbc-4225-89be-43112cbb04e3" />

**Comment by ioanarusiczki** [comment]
All looking good yey ,  Change theme and the option bar looking better below Change language.

Android icon on detail page was pink now it stays green , the rating star is no longer black 🎉

One thing is different for ratings - the "Star rating saved" moved towards Write a review button , dark mode makes this a bit more obvious 

<img width="1670" height="442" alt="Image" src="https://github.com/user-attachments/assets/7be8760e-13a6-47b3-a679-fab7ec6e610f" />

currently on AMO stage 

<img width="1672" height="397" alt="Image" src="https://github.com/user-attachments/assets/ea8118e1-93d1-402b-b674-ebdebbdb6f1f" />

And I forgot to mention the grey txt on More information section from the detail pages 

<img width="1738" height="545" alt="Image" src="https://github.com/user-attachments/assets/479901ab-cdfb-4f97-a2f9-4654013c7ad8" />

So if you need followups I could file these separately @nothingrandom let me know (gotta step out today but I'll be back tomorrow morning)

**Comment by nothingrandom** [comment]
Thank you @ioanarusiczki. If we can agree that this is fine for release as is, spinning these into new follow up tickets would be preferred. If you think these block this feature from being shipped, then we'll keep it here and reopen the issue.

**Comment by abyrne-moz** [comment]
I don't think this should block shipping what we have for now, and working on these fixes in the next sprint (I'd rather not hold them up any longer than that)

**Comment by ioanarusiczki** [comment]
Definitely not blockers, I filed followups and in case I'll notice other stuffs I'll add comments to those.
